### Task 1 - identify the numbers of rows and columns in it and find any null values are present 

In [2]:
import pandas as pd

In [3]:
data = pd.read_csv("DS1_C4_S3_Loan_Data_Challenge.csv")

In [4]:
data

Loan_ID  Gender Married Dependents     Education Self_Employed  \
0    LP001002    Male      No          0      Graduate            No   
1    LP001003    Male     Yes          1      Graduate            No   
2    LP001005    Male     Yes          0      Graduate           Yes   
3    LP001006    Male     Yes          0  Not Graduate            No   
4    LP001008    Male      No          0      Graduate            No   
..        ...     ...     ...        ...           ...           ...   
609  LP002978  Female      No          0      Graduate            No   
610  LP002979    Male     Yes         3+      Graduate            No   
611  LP002983    Male     Yes          1      Graduate            No   
612  LP002984    Male     Yes          2      Graduate            No   
613  LP002990  Female      No          0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0         146               360   
1               4583             1508.0         128               360   
2               3000                0.0          66               360   
3               2583             2358.0         120               360   
4               6000                0.0         141               360   
..               ...                ...         ...               ...   
609             2900                0.0          71               360   
610             4106                0.0          40               180   
611             8072              240.0         253               360   
612             7583                0.0         187               360   
613             4583                0.0         133               360   

     Credit_History Property_Area Loan_Status  
0                 1         Urban           Y  
1                 1         Rural           N  
2                 1         Urban           Y  
3                 1         Urban           Y  
4                 1         Urban           Y  
..              ...           ...         ...  
609               1         Rural           Y  
610               1         Rural           Y  
611               1         Urban           Y  
612               1         Urban           Y  
613               0     Semiurban           N  

[614 rows x 13 columns]

In [5]:
data.isnull().sum()

Loan_ID              0
Gender               0
Married              0
Dependents           0
Education            0
Self_Employed        0
ApplicantIncome      0
CoapplicantIncome    0
LoanAmount           0
Loan_Amount_Term     0
Credit_History       0
Property_Area        0
Loan_Status          0
dtype: int64

Interpretation - There is no null value present in the data . 

### Task 2 - Display the list of female customers ,who are self-employeed and salary greater than 4000 

In [6]:
gen = data[(data["Gender"]=="Female") & (data["Self_Employed"]=="Yes") & (data["ApplicantIncome"]>4000)]

In [7]:
gen

Loan_ID  Gender Married Dependents     Education Self_Employed  \
54   LP001186  Female     Yes          1      Graduate           Yes   
113  LP001392  Female      No          1      Graduate           Yes   
353  LP002142  Female     Yes          0      Graduate           Yes   
370  LP002194  Female      No          0      Graduate           Yes   
404  LP002301  Female      No          0      Graduate           Yes   
430  LP002377  Female      No          1      Graduate           Yes   
439  LP002407  Female     Yes          0  Not Graduate           Yes   
463  LP002489  Female      No          1  Not Graduate           Yes   
493  LP002582  Female      No          0  Not Graduate           Yes   
534  LP002731  Female      No          0  Not Graduate           Yes   
561  LP002813  Female     Yes          1      Graduate           Yes   
613  LP002990  Female      No          0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
54             11500                0.0         286               360   
113             7451                0.0         146               360   
353             5500                0.0         105               360   
370            15759                0.0          55               360   
404             7441                0.0         194               360   
430             8624                0.0         150               360   
439             7142                0.0         138               360   
463             5191                0.0         132               360   
493            17263                0.0         225               360   
534            18165                0.0         125               360   
561            19484                0.0         600               360   
613             4583                0.0         133               360   

     Credit_History Property_Area Loan_Status  
54                0         Urban           N  
113               1     Semiurban           Y  
353               0         Rural           N  
370               1     Semiurban           Y  
404               1         Rural           N  
430               1     Semiurban           Y  
439               1         Rural           Y  
463               1     Semiurban           Y  
493               1     Semiurban           Y  
534               1         Urban           Y  
561               1     Semiurban           Y  
613               0     Semiurban           N

Interpretation - Here i have displayed  the records ,list of female customers ,who are self-employeed and salary greater than 4000

### Task 3 - Identify those male customers who hold gratuated degree and having loan status as 'Y' (yes)

In [8]:
male = data[(data["Gender"]=="Male") & (data["Education"]=="Graduate") & (data["Loan_Status"]=="Y")]

In [9]:
male

Loan_ID Gender Married Dependents Education Self_Employed  \
0    LP001002   Male      No          0  Graduate            No   
2    LP001005   Male     Yes          0  Graduate           Yes   
4    LP001008   Male      No          0  Graduate            No   
5    LP001011   Male     Yes          2  Graduate           Yes   
8    LP001018   Male     Yes          2  Graduate            No   
..        ...    ...     ...        ...       ...           ...   
606  LP002961   Male     Yes          1  Graduate            No   
608  LP002974   Male     Yes          0  Graduate            No   
610  LP002979   Male     Yes         3+  Graduate            No   
611  LP002983   Male     Yes          1  Graduate            No   
612  LP002984   Male     Yes          2  Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0         146               360   
2               3000                0.0          66               360   
4               6000                0.0         141               360   
5               5417             4196.0         267               360   
8               4006             1526.0         168               360   
..               ...                ...         ...               ...   
606             3400             2500.0         173               360   
608             3232             1950.0         108               360   
610             4106                0.0          40               180   
611             8072              240.0         253               360   
612             7583                0.0         187               360   

     Credit_History Property_Area Loan_Status  
0                 1         Urban           Y  
2                 1         Urban           Y  
4                 1         Urban           Y  
5                 1         Urban           Y  
8                 1         Urban           Y  
..              ...           ...         ...  
606               1     Semiurban           Y  
608               1         Rural           Y  
610               1         Rural           Y  
611               1         Urban           Y  
612               1         Urban           Y  

[279 rows x 13 columns]

In [10]:
male.shape

(279, 13)

Interpretation - Here i have displayed those male customers who hold gratuated degree and having loan status as 'Y' (yes) and also the count of these customers is 279

### Task 4  

In [11]:
data[(data['Loan_Status']=="Y") & (data['Dependents'] >= "3") & (data['Property_Area']=="Rural")]

Loan_ID  Gender Married Dependents     Education Self_Employed  \
126  LP001448    Male     Yes         3+      Graduate            No   
213  LP001715    Male     Yes         3+  Not Graduate           Yes   
255  LP001846  Female      No         3+      Graduate            No   
324  LP002065    Male     Yes         3+      Graduate            No   
352  LP002141    Male     Yes         3+      Graduate            No   
376  LP002219    Male     Yes         3+      Graduate            No   
390  LP002255    Male      No         3+      Graduate            No   
391  LP002262    Male     Yes         3+      Graduate            No   
481  LP002536    Male     Yes         3+  Not Graduate            No   
515  LP002659    Male     Yes         3+      Graduate            No   
522  LP002692    Male     Yes         3+      Graduate           Yes   
539  LP002740    Male     Yes         3+      Graduate            No   
610  LP002979    Male     Yes         3+      Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
126            23803                0.0         370               360   
213             5703                0.0         130               360   
255             3083                0.0         255               360   
324            15000                0.0         300               360   
352             2666             2083.0          95               360   
376             8750             4996.0         130               360   
390             9167                0.0         185               360   
391             9504                0.0         275               360   
481             3095                0.0         113               360   
515             3466             3428.0         150               360   
522             5677             1424.0         100               360   
539             6417                0.0         157               180   
610             4106                0.0          40               180   

     Credit_History Property_Area Loan_Status  
126               1         Rural           Y  
213               1         Rural           Y  
255               1         Rural           Y  
324               1         Rural           Y  
352               1         Rural           Y  
376               1         Rural           Y  
390               1         Rural           Y  
391               1         Rural           Y  
481               1         Rural           Y  
515               1         Rural           Y  
522               1         Rural           Y  
539               1         Rural           Y  
610               1         Rural           Y

Interpretation - Here i have displayed the customer details whose loan status = Y , dependents = 3+ , Property_Area = rural and i didn't create any separate dataframe to display the details  

### Task 5 

In [12]:
warner = data[(data["Gender"]=="Male") & (data["Education"]=="Graduate")  & (data["Property_Area"]=="Urban")]
warner

Loan_ID Gender Married Dependents Education Self_Employed  \
0    LP001002   Male      No          0  Graduate            No   
2    LP001005   Male     Yes          0  Graduate           Yes   
4    LP001008   Male      No          0  Graduate            No   
5    LP001011   Male     Yes          2  Graduate           Yes   
8    LP001018   Male     Yes          2  Graduate            No   
..        ...    ...     ...        ...       ...           ...   
594  LP002938   Male     Yes          0  Graduate           Yes   
599  LP002948   Male     Yes          2  Graduate            No   
602  LP002953   Male     Yes         3+  Graduate            No   
611  LP002983   Male     Yes          1  Graduate            No   
612  LP002984   Male     Yes          2  Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0         146               360   
2               3000                0.0          66               360   
4               6000                0.0         141               360   
5               5417             4196.0         267               360   
8               4006             1526.0         168               360   
..               ...                ...         ...               ...   
594            16120                0.0         260               360   
599             5780                0.0         192               360   
602             5703                0.0         128               360   
611             8072              240.0         253               360   
612             7583                0.0         187               360   

     Credit_History Property_Area Loan_Status  
0                 1         Urban           Y  
2                 1         Urban           Y  
4                 1         Urban           Y  
5                 1         Urban           Y  
8                 1         Urban           Y  
..              ...           ...         ...  
594               1         Urban           Y  
599               1         Urban           Y  
602               1         Urban           Y  
611               1         Urban           Y  
612               1         Urban           Y  

[133 rows x 13 columns]

Interpretation - Warner kim will definetly get the loan .

### Task 6 

In [13]:
task_five = data[(data["Married"]=="No") & (data["Self_Employed"]=="Yes") & (data["Property_Area"]!="Urban")
                 & (data["Loan_Status"]=="N")]

In [14]:
task_five

Loan_ID  Gender Married Dependents     Education Self_Employed  \
32   LP001097    Male      No          1      Graduate           Yes   
107  LP001370    Male      No          0  Not Graduate           Yes   
286  LP001925  Female      No          0      Graduate           Yes   
404  LP002301  Female      No          0      Graduate           Yes   
410  LP002318  Female      No          1  Not Graduate           Yes   
449  LP002444    Male      No          1  Not Graduate           Yes   
613  LP002990  Female      No          0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
32              4692                0.0         106               360   
107             7333                0.0         120               360   
286             2600             1717.0          99               300   
404             7441                0.0         194               360   
410             3867                0.0          62               360   
449             2769             1542.0         190               360   
613             4583                0.0         133               360   

     Credit_History Property_Area Loan_Status  
32                1         Rural           N  
107               1         Rural           N  
286               1     Semiurban           N  
404               1         Rural           N  
410               1     Semiurban           N  
449               1     Semiurban           N  
613               0     Semiurban           N

In [15]:
gender_grp = task_five.groupby(["Gender"])

In [16]:
gender_grp

In [17]:
for x,y in gender_grp:
    print(x)
    
    print(y)

Female
      Loan_ID  Gender Married Dependents     Education Self_Employed  \
286  LP001925  Female      No          0      Graduate           Yes   
404  LP002301  Female      No          0      Graduate           Yes   
410  LP002318  Female      No          1  Not Graduate           Yes   
613  LP002990  Female      No          0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
286             2600             1717.0          99               300   
404             7441                0.0         194               360   
410             3867                0.0          62               360   
613             4583                0.0         133               360   

     Credit_History Property_Area Loan_Status  
286               1     Semiurban           N  
404               1         Rural           N  
410               1     Semiurban           N  
613               0     Semiurban           N  
Male
      Loan_ID Gender 

C:\Users\win 10\AppData\Local\Temp\ipykernel_18832\1236665136.py:1: FutureWarning: In a future version of pandas, a length 1 tuple will be returned when iterating over a groupby with a grouper equal to a list of length 1. Don't supply a list with a single grouper to avoid this warning.
  for x,y in gender_grp:


Interpretation - There are totally 7 customers who satisfy the above conditions and there are 4 Female and 3 male. 

### Task  7 

In [18]:
avg_income = data.groupby(by=["Property_Area"])["ApplicantIncome"].mean()

In [19]:
avg_income

Property_Area
Rural        5554.083799
Semiurban    5292.261803
Urban        5398.247525
Name: ApplicantIncome, dtype: float64

Interpretation - The above i have displayed the average income of the customers by property area 

### Task 8 

In [20]:
task_eight = data[(data["Property_Area"]!="Urban") & (data["ApplicantIncome"]>4000) & (data["CoapplicantIncome"]>1000) 
                  & (data["Loan_Status"]=="N")]

In [21]:
task_eight

Loan_ID  Gender Married Dependents     Education Self_Employed  \
1    LP001003    Male     Yes          1      Graduate            No   
9    LP001020    Male     Yes          1      Graduate            No   
34   LP001100    Male      No         3+      Graduate            No   
65   LP001225    Male     Yes          0      Graduate            No   
83   LP001273    Male     Yes          0      Graduate            No   
118  LP001421    Male     Yes          0      Graduate            No   
129  LP001465    Male     Yes          0      Graduate            No   
139  LP001493    Male     Yes          2  Not Graduate            No   
140  LP001497    Male     Yes          2      Graduate            No   
148  LP001519  Female      No          0      Graduate            No   
177  LP001610    Male     Yes         3+      Graduate            No   
258  LP001859    Male     Yes          0      Graduate            No   
298  LP001955  Female      No          0      Graduate            No   
325  LP002067    Male     Yes          1      Graduate           Yes   
351  LP002140    Male      No          0      Graduate            No   
369  LP002191    Male     Yes          0      Graduate            No   
499  LP002602    Male      No          0      Graduate            No   
503  LP002618    Male     Yes          1  Not Graduate            No   
514  LP002652    Male      No          0      Graduate            No   
518  LP002683    Male      No          0      Graduate            No   
524  LP002697    Male      No          0      Graduate            No   
573  LP002862    Male     Yes          2  Not Graduate            No   
585  LP002912    Male     Yes          1      Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
1               4583             1508.0         128               360   
9              12841            10968.0         349               360   
34             12500             3000.0         320               360   
65              5726             4595.0         258               360   
83              6000             2250.0         265               360   
118             5568             2142.0         175               360   
129             6080             2569.0         182               360   
139             4200             1430.0         129               360   
140             5042             2083.0         185               360   
148            10000             1666.0         225               360   
177             5516            11300.0         495               360   
258            14683             2100.0         304               360   
298             5000             2541.0         151               480   
325             8666             4983.0         376               360   
351             8750             4167.0         308               360   
369            19730             5266.0         570               360   
499             6283             4416.0         209               360   
503             4050             5302.0         138               360   
514             5815             3666.0         311               360   
518             4683             1915.0         185               360   
524             4680             2087.0         146               360   
573             6125             1625.0         187               480   
585             4283             3000.0         172                84   

     Credit_History Property_Area Loan_Status  
1                 1         Rural           N  
9                 1     Semiurban           N  
34                1         Rural           N  
65                1     Semiurban           N  
83                1     Semiurban           N  
118               1         Rural           N  
129               1         Rural           N  
139               1         Rural           N  
140               1         Rural           N  
148               1         Rural           

In [22]:
task_eight['Loan_Status']= task_eight['Loan_Status'].str.replace("N", "Y")

C:\Users\win 10\AppData\Local\Temp\ipykernel_18832\392957923.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  task_eight['Loan_Status']= task_eight['Loan_Status'].str.replace("N", "Y")


In [24]:
task_eight

Loan_ID  Gender Married Dependents     Education Self_Employed  \
1    LP001003    Male     Yes          1      Graduate            No   
9    LP001020    Male     Yes          1      Graduate            No   
34   LP001100    Male      No         3+      Graduate            No   
65   LP001225    Male     Yes          0      Graduate            No   
83   LP001273    Male     Yes          0      Graduate            No   
118  LP001421    Male     Yes          0      Graduate            No   
129  LP001465    Male     Yes          0      Graduate            No   
139  LP001493    Male     Yes          2  Not Graduate            No   
140  LP001497    Male     Yes          2      Graduate            No   
148  LP001519  Female      No          0      Graduate            No   
177  LP001610    Male     Yes         3+      Graduate            No   
258  LP001859    Male     Yes          0      Graduate            No   
298  LP001955  Female      No          0      Graduate            No   
325  LP002067    Male     Yes          1      Graduate           Yes   
351  LP002140    Male      No          0      Graduate            No   
369  LP002191    Male     Yes          0      Graduate            No   
499  LP002602    Male      No          0      Graduate            No   
503  LP002618    Male     Yes          1  Not Graduate            No   
514  LP002652    Male      No          0      Graduate            No   
518  LP002683    Male      No          0      Graduate            No   
524  LP002697    Male      No          0      Graduate            No   
573  LP002862    Male     Yes          2  Not Graduate            No   
585  LP002912    Male     Yes          1      Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
1               4583             1508.0         128               360   
9              12841            10968.0         349               360   
34             12500             3000.0         320               360   
65              5726             4595.0         258               360   
83              6000             2250.0         265               360   
118             5568             2142.0         175               360   
129             6080             2569.0         182               360   
139             4200             1430.0         129               360   
140             5042             2083.0         185               360   
148            10000             1666.0         225               360   
177             5516            11300.0         495               360   
258            14683             2100.0         304               360   
298             5000             2541.0         151               480   
325             8666             4983.0         376               360   
351             8750             4167.0         308               360   
369            19730             5266.0         570               360   
499             6283             4416.0         209               360   
503             4050             5302.0         138               360   
514             5815             3666.0         311               360   
518             4683             1915.0         185               360   
524             4680             2087.0         146               360   
573             6125             1625.0         187               480   
585             4283             3000.0         172                84   

     Credit_History Property_Area Loan_Status  
1                 1         Rural           Y  
9                 1     Semiurban           Y  
34                1         Rural           Y  
65                1     Semiurban           Y  
83                1     Semiurban           Y  
118               1         Rural           Y  
129               1         Rural           Y  
139               1         Rural           Y  
140               1         Rural           Y  
148               1         Rural           

Interpretation - here i have updated the loan status from "N" to "Y"of customers who satisfy the above filters 

### Task 9 

In [25]:
avg_count = data[(data["Dependents"]=="3+") & (data["Property_Area"]=="Urban") & (data["Loan_Status"]=="Y")]

In [26]:
avg_count

Loan_ID Gender Married Dependents     Education Self_Employed  \
61   LP001206   Male     Yes         3+      Graduate            No   
68   LP001238   Male     Yes         3+  Not Graduate           Yes   
171  LP001585   Male     Yes         3+      Graduate            No   
267  LP001882   Male     Yes         3+      Graduate            No   
295  LP001949   Male     Yes         3+      Graduate            No   
461  LP002484   Male     Yes         3+      Graduate            No   
531  LP002720   Male     Yes         3+      Graduate            No   
602  LP002953   Male     Yes         3+      Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
61              3029                0.0          99               360   
68              7100                0.0         125                60   
171            51763                0.0         700               300   
267             4333             1811.0         160               360   
295             4416             1250.0         110               360   
461             7740                0.0         128               180   
531             4281                0.0         100               360   
602             5703                0.0         128               360   

     Credit_History Property_Area Loan_Status  
61                1         Urban           Y  
68                1         Urban           Y  
171               1         Urban           Y  
267               0         Urban           Y  
295               1         Urban           Y  
461               1         Urban           Y  
531               1         Urban           Y  
602               1         Urban           Y

In [27]:
avg = avg_count["ApplicantIncome"].mean()

In [28]:
avg

11045.625

In [29]:
avg_count.shape

(8, 13)

Interpretation - There are totally 8 customer who satisfy the above conditions and the average income of those customers is 11045.625

### Task 10 

In [30]:
area = data[data["Property_Area"]=="Urban"]
area

Loan_ID  Gender Married Dependents     Education Self_Employed  \
0    LP001002    Male      No          0      Graduate            No   
2    LP001005    Male     Yes          0      Graduate           Yes   
3    LP001006    Male     Yes          0  Not Graduate            No   
4    LP001008    Male      No          0      Graduate            No   
5    LP001011    Male     Yes          2      Graduate           Yes   
..        ...     ...     ...        ...           ...           ...   
600  LP002949  Female      No         3+      Graduate           Yes   
602  LP002953    Male     Yes         3+      Graduate            No   
605  LP002960    Male     Yes          0  Not Graduate            No   
611  LP002983    Male     Yes          1      Graduate            No   
612  LP002984    Male     Yes          2      Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0         146               360   
2               3000                0.0          66               360   
3               2583             2358.0         120               360   
4               6000                0.0         141               360   
5               5417             4196.0         267               360   
..               ...                ...         ...               ...   
600              416            41667.0         350               180   
602             5703                0.0         128               360   
605             2400             3800.0         146               180   
611             8072              240.0         253               360   
612             7583                0.0         187               360   

     Credit_History Property_Area Loan_Status  
0                 1         Urban           Y  
2                 1         Urban           Y  
3                 1         Urban           Y  
4                 1         Urban           Y  
5                 1         Urban           Y  
..              ...           ...         ...  
600               0         Urban           N  
602               1         Urban           Y  
605               1         Urban           N  
611               1         Urban           Y  
612               1         Urban           Y  

[202 rows x 13 columns]

In [31]:
ar = area["ApplicantIncome"].mean()
ar

5398.247524752475

In [32]:
s = data[(data["Property_Area"]!="Urban") & (data["Dependents"]=="3+") & (data["ApplicantIncome"]>5398.247)]

In [33]:
s

Loan_ID Gender Married Dependents     Education Self_Employed  \
34   LP001100   Male      No         3+      Graduate            No   
126  LP001448   Male     Yes         3+      Graduate            No   
155  LP001536   Male     Yes         3+      Graduate            No   
177  LP001610   Male     Yes         3+      Graduate            No   
213  LP001715   Male     Yes         3+  Not Graduate           Yes   
324  LP002065   Male     Yes         3+      Graduate            No   
376  LP002219   Male     Yes         3+      Graduate            No   
390  LP002255   Male      No         3+      Graduate            No   
391  LP002262   Male     Yes         3+      Graduate            No   
409  LP002317   Male     Yes         3+      Graduate            No   
522  LP002692   Male     Yes         3+      Graduate           Yes   
539  LP002740   Male     Yes         3+      Graduate            No   
557  LP002795   Male     Yes         3+      Graduate           Yes   
574  LP002863   Male     Yes         3+      Graduate            No   
592  LP002933   Male      No         3+      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
34             12500             3000.0         320               360   
126            23803                0.0         370               360   
155            39999                0.0         600               180   
177             5516            11300.0         495               360   
213             5703                0.0         130               360   
324            15000                0.0         300               360   
376             8750             4996.0         130               360   
390             9167                0.0         185               360   
391             9504                0.0         275               360   
409            81000                0.0         360               360   
522             5677             1424.0         100               360   
539             6417                0.0         157               180   
557            10139                0.0         260               360   
574             6406                0.0         150               360   
592             9357                0.0         292               360   

     Credit_History Property_Area Loan_Status  
34                1         Rural           N  
126               1         Rural           Y  
155               0     Semiurban           Y  
177               0     Semiurban           N  
213               1         Rural           Y  
324               1         Rural           Y  
376               1         Rural           Y  
390               1         Rural           Y  
391               1         Rural           Y  
409               0         Rural           N  
522               1         Rural           Y  
539               1         Rural           Y  
557               1     Semiurban           Y  
574               1     Semiurban           N  
592               1     Semiurban           Y

1) I have infered something from the above task that the bank gave loan for the customers who have greater than 3+ Dependents 

In [34]:
A = s["ApplicantIncome"].mean()

In [35]:
A

16595.866666666665

2)These are the average applicantincome of the customers who are from non-Urban area